lineapy/data/graph.py

```diff
@@ -34,6 +34,7 @@ def __init__(self, nodes: List[Node], session_context: SessionContext):
                 (parent_id, node.id)
                 for node in nodes
                 for parent_id in node.parents()
+                if parent_id in set(self.ids.keys())
```

Comment:
Need this to make sure the graph is self-contained within the list of input nodes.